### PR TITLE
dpb-30, dpb-27

### PR DESCRIPTION
**Pull request by Tchander**
https://trello.com/c/uu8TMDRP/30-30-add-timer-to-draft
https://trello.com/c/qwsVQhXn/27-27-add-disable-state-for-pick-ban-button

### CODE REVIEW

File: src/components/ConfirmationHero.vue [hunk 1]
```diff
@@ -44,6 +46,10 @@ interface Props {
 
 const props = defineProps<Props>();
 
+const isButtonDisabled = computed<boolean>(() => {
```
Comment (Vterebenin) [inline src/components/ConfirmationHero.vue]: overengineering

Comment (Vterebenin) [reply src/components/ConfirmationHero.vue]: simply use props on the template

File: src/components/DraftList.vue [hunk 2]
```diff
@@ -44,14 +51,43 @@
...
+
+const mainTime = computed<number>(() => {
+  if (props.team !== TeamList.RADIANT) return direMainTime.value;
+  return radiantMainTime.value;
+});
+
+const transformTime = (time: number) => {
+  const integer = Math.floor(time / SECONDS_IN_MINUTE);
+  const fractional = Math.floor(time % SECONDS_IN_MINUTE);
+  return `${integer}:${fractional >= 10 ? fractional : `0${fractional}`}`;
```
Comment (Vterebenin) [inline src/components/DraftList.vue:94]: 10 is a magic number

File: src/components/DraftSection.vue [hunk 3]
```diff
@@ -128,11 +134,95 @@ const heroChosen = (hero: IHero | null) => {
...
 
+const timerStart = () => {
+  window.clearInterval(timerStore.radiantTimerInterval);
+  window.clearInterval(timerStore.direTimerInterval);
+  timerStore.setTimeByDefault(TimerMapper.RADIANT_MAIN_TIME);
+  timerStore.setTimeByDefault(TimerMapper.DIRE_MAIN_TIME);
+
+  if (state.roundCounter > DEFAULT_NUMBER_OF_ROUNDS) return;
+
+  if (
```
Comment (Vterebenin) [inline src/components/DraftSection.vue:174]: we need to refactor this somehow; let's discuss the data structures and break this if statement into smaller atomic functions

File: src/components/DraftSection.vue [hunk 4]
```diff
@@ -128,11 +134,95 @@ const heroChosen = (hero: IHero | null) => {
...
+    FIRST_PICK_PICK_ROUNDS.includes(state.roundCounter) ||
+    FIRST_PICK_BAN_ROUNDS.includes(state.roundCounter)
+  ) {
+    if (configStore.isRadiantFirstPick) {
+      timerStore.radiantTimerInterval = window.setInterval(() => {
```
Comment (Vterebenin) [inline src/components/DraftSection.vue]: `setInterval` instead of `window.setInterval`

File: src/components/DraftSection.vue [hunk 5]
```diff
@@ -128,11 +134,95 @@ const heroChosen = (hero: IHero | null) => {
...
+        if (timerStore.radiantMainTime === 0) {
+          if (timerStore.radiantReserveTime === 0) {
+            heroChosen(null);
```
Comment (Vterebenin) [inline src/components/DraftSection.vue]: why null?

File: src/components/DraftSection.vue [hunk 6]
```diff
@@ -128,11 +134,95 @@ const heroChosen = (hero: IHero | null) => {
...
+          } else {
+            timerStore.decreaseTimer(TimerMapper.RADIANT_RESERVE_TIME);
+          }
+        } else {
+          timerStore.decreaseTimer(TimerMapper.RADIANT_MAIN_TIME);
+        }
+      }, 1000);
```
Comment (Vterebenin) [inline src/components/DraftSection.vue]: 1000 magic number

File: src/stores/config.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import { StoreId } from '@/enum/stores';
+import { IConfigState } from '@/types';
+
+export const useConfigStore = defineStore(StoreId.CONFIG, {
+  state: (): IConfigState => ({
+    isRadiantFirstPick: false,
+    isTimer: false,
+  }),
+
+  getters: {},
```
Comment (Vterebenin) [inline src/stores/config.ts]: remove getters

**Review by Vterebenin — CHANGES_REQUESTED** [review]

File: src/components/DraftSection.vue (hunk 4 above)
Comment (Tchander) [reply src/components/DraftSection.vue]: https://stackoverflow.com/questions/53189729/typescript-error-setinterval-type-timer-is-not-assignable-to-type-number
I had this problem. That's why I used window.setInterval

File: src/components/DraftSection.vue (hunk 5 above)
Comment (Tchander) [reply src/components/DraftSection.vue]: If the player didn't have time to ban hero then I put null against the hero to array of banned heroes in certain position

File: src/components/DraftSection.vue (hunk 4 above)
Comment (Tchander) [reply src/components/DraftSection.vue]: image

**Review by Vterebenin — COMMENTED** [review]
LGTM

**Pull request merged**